docs/implementation-status.md

```diff
@@ -0,0 +1,245 @@
+# Implementation Status
+
+Caffeine is an implementation of the Parallel Runtime Interface for Fortran (PRIF). This document
+outlines the implementation status in Caffeine of the features defined in the
+[latest PRIF revision](https://dx.doi.org/10.25344/S4WG64). Caffeine contains interfaces for all
+of the PRIF procedures and are callable, but some procedure definitions are unimplemented. For
+more details about the implementation of the various PRIF features, please see the
+following sections:
+
+- [Named Constants](#Named-Constants)
+- [`stat` and `errmsg` support](#stat-and-errmsg-support) - NO support yet
+- [Program Startup and Shutdown](#Program-Startup-and-Shutdown)
+- [Image Queries](#Image-Queries)
+- [Storage Management](#Storage-Management)
+- [Coarray Queries](#Coarray-Queries)
+- [Continguous Coarray Access](#Continguous-Coarray-Access)
+- [Strided Coarray Access](#Strided-Coarray-Access)
+- [SYNC Statements](#SYNC-Statements)
+- [Locks and Unlocks](#Locks-and-Unlocks)
+- [Critical](#Critical)
+- [Events and Notifications](#Events-and-Notifications)
+- [Teams](#teams)
+- [Collectives](#Collectives)
+- [Atomic Memory Operations](#Atomic-Memory-Operations)
+
+TODO: For features with all no support, flesh out table, but comment it out
+
+## Named Constants
+
+Caffeine contains definitions for all of the PRIF-relevant constants from ISO_FORTRAN_ENV and for
+all of the PRIF-specific constants.
+
+## `stat` and `errmsg` support
+
+Many PRIF procedures have optional arguments `stat`, `errmsg`, and `errmsg_alloc`. There is no
+support for these optional arguments in Caffeine yet and this is not represented in the tables
+in the following sections.
+
+
+## Program Startup and Shutdown
+### Support = partial
+
+| Procedure | Status | Notes |
+|-----------|--------|-------|
+| `prif_init` | yes | n/a |
+| `prif_stop`, `prif_error_stop` | partial | Missing support for `quiet=.true.` |
+| `prif_fail_image` | no | n/a |
+
+---
+
+## Image Queries
+### Support = partial
+
+| Procedure | Status | Notes |
+|-----------|--------|-------|
+| `prif_num_images`    | yes | n\a |
+| `prif_num_images_with_team`, `prif_num_images_with_team_number`  | no | n\a |
+| `prif_this_image_no_coarray` | yes | n\a |
+| `prif_this_image_with_coarray`, `prif_this_image_with_dim`  | no | n\a |
+| `prif_failed_images` | no | n/a |
+| `prif_stopped_images` | no | n/a |
+| `prif_image_status` | no | n/a |
+
+---
+
+## Storage Management
+### Support = partial (no support for coarray aliases)
+
+| Procedure | Status | Notes |
+|-----------|--------|-------|
+| `prif_allocate_coarray`    | yes | n/a |
+| `prif_allocate`    | yes | n/a |
+| `prif_deallocate_coarray`    | partial | no `final_func` arg support |
+| `prif_deallocate`    | yes | n/a |
+| `prif_alias_create`    | no | n/a |
```

Comment:
On a cosmetic note, even in the absence of color we could potentially still make the Status column "pop" a bit more, eg:

| Procedure | Status | Notes |
|-----------|--------|-------|
| `prif_allocate_coarray`    | **YES** |  |
| `prif_allocate`    | **YES** |  |
| `prif_deallocate_coarray`    | *partial* | no `final_func` arg support |
| `prif_deallocate`    | **YES** |  |
| `prif_alias_create`    | no |  |
| `prif_alias_destroy`    | no | |

However if this seems like too much hassle to maintain then we could keep the current convention.

```suggestion
| `prif_deallocate_coarray`    | *partial* | no `final_func` arg support |
| `prif_deallocate`    | **YES** | n/a |
| `prif_alias_create`    | no | n/a |
```